flutter/lib/src/screenshot/screenshot_attachment.dart

```diff
@@ -0,0 +1,63 @@
+import 'dart:async';
+import 'dart:typed_data';
+import 'dart:ui' as ui show ImageByteFormat;
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:sentry/sentry.dart';
+
+import '../sentry_widget.dart';
+
+class ScreenshotAttachment implements SentryAttachment {
```

Comment:
Maybe this class should just be a factory method from `SentryAttachment`
Similar to `SentryAttachment.fromScreenshot(binary...)`

Reply:
The `ScreenshotAttachment ` is lazy, only creating the screenshot when it's `data` field is accessed, that's why it is its own class. If we move the screenshot creation outside, for example when the attachments are processed, than we can just use a custom ctor. @ueman I took this approach from you, any particular reason behind making the data property triggering the screenshot?

Reply:
I think the reasoning was that you can add the attachment just once, and then on every new event to Sentry you have a new current screenshot. At that time there was no way to remove attachments, and having the same image for all events doesn't make a lot of sense.

If that's now possible, you don't even need the `ScreenshotAttachment` but can do the screenshot somewhere else and then you can use a regular attachment to add it to an event.

Reply:
Alright, then let's change this behaviour.

Reply:
Please remember to remove the screenshot after the event was sent 😅

Reply:
Yes removing attachments is still depending on https://github.com/getsentry/sentry-dart/issues/510
Planned for v7, so we can refactor this if necessary

Reply:
Don't know what removing means in this context? We are only adding the attachment to the collection right before it gets passed on as an envelope. It's not added to the scopes attachments.